remote-messaging/remote-messaging-impl/src/main/java/com/duckduckgo/remote/messaging/impl/di/RemoteMessagingModule.kt

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (c) 2021 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.remote.messaging.impl.di
+
+import android.content.Context
+import androidx.room.Room
+import com.duckduckgo.app.global.AppUrl
+import com.duckduckgo.remote.messaging.impl.mappers.JsonRemoteMessageMapper
+import com.duckduckgo.remote.messaging.impl.mappers.JsonRulesMapper
+import com.duckduckgo.remote.messaging.impl.RealRemoteMessagingConfigDownloader
+import com.duckduckgo.remote.messaging.impl.RemoteMessagingConfigDownloader
+import com.duckduckgo.di.scopes.AppScope
+import com.duckduckgo.remote.messaging.impl.RealRemoteMessagingConfigProcessor
+import com.duckduckgo.remote.messaging.impl.RemoteMessagingConfigProcessor
+import com.duckduckgo.remote.messaging.impl.mappers.RemoteMessagingConfigJsonMapper
+import com.duckduckgo.remote.messaging.impl.network.RemoteMessagingService
+import com.duckduckgo.remote.messaging.store.ALL_MIGRATIONS
+import com.duckduckgo.remote.messaging.store.LocalRemoteMessagingConfigRepository
+import com.duckduckgo.remote.messaging.store.RemoteMessagingConfigRepository
+import com.duckduckgo.remote.messaging.store.RemoteMessagingDatabase
+import com.squareup.anvil.annotations.ContributesTo
+import com.squareup.moshi.Moshi
+import dagger.Module
+import dagger.Provides
+import dagger.SingleInstanceIn
+import okhttp3.OkHttpClient
+import retrofit2.Retrofit
+import retrofit2.converter.moshi.MoshiConverterFactory
+import javax.inject.Named
+
+@Module
```

Comment:
nit: Just curious, why having all this dagger modules as oppose to use `ContributesBinding` etc directly in the `Real...` implementation classes.
It creates a bit of overhead, eg. I add a new parameter to `RealRemoteMessagingConfigDownloader` and now I need to modify both the `RealRemoteMessagingConfigDownloader` and the `DomainModule` files.